src/main.rs

```diff
@@ -15,30 +15,73 @@ use std::path::PathBuf;
 pub type Cost = NotNan<f64>;
 pub const INFINITY: Cost = unsafe { NotNan::new_unchecked(std::f64::INFINITY) };
 
-fn main() {
-    env_logger::init();
+struct ExtractorDetail {
+    extractor: Box<dyn Extractor>,
+    is_dag_optimal: bool,
+    is_tree_optimal: bool,
+}
 
-    let extractors: IndexMap<&str, Box<dyn Extractor>> = [
-        ("bottom-up", extract::bottom_up::BottomUpExtractor.boxed()),
+fn extractors() -> IndexMap<&'static str, ExtractorDetail> {
+    let extractors: IndexMap<&'static str, ExtractorDetail> = [
         (
-            "faster-bottom-up",
-            extract::faster_bottom_up::FasterBottomUpExtractor.boxed(),
+            "bottom-up",
+            ExtractorDetail {
+                extractor: extract::bottom_up::BottomUpExtractor.boxed(),
+                is_dag_optimal: false,
+                is_tree_optimal: true,
+            },
         ),
         (
-            "greedy-dag",
-            extract::greedy_dag::GreedyDagExtractor.boxed(),
+            "faster-bottom-up",
+            ExtractorDetail {
+                extractor: extract::faster_bottom_up::FasterBottomUpExtractor.boxed(),
+                is_dag_optimal: false,
+                is_tree_optimal: true,
+            },
         ),
         (
             "faster-greedy-dag",
-            extract::faster_greedy_dag::FasterGreedyDagExtractor.boxed(),
+            ExtractorDetail {
+                extractor: extract::faster_greedy_dag::FasterGreedyDagExtractor.boxed(),
+                is_dag_optimal: false,
+                is_tree_optimal: false,
+            },
         ),
         (
+            "greedy-dag",
+            ExtractorDetail {
+                extractor: extract::greedy_dag::GreedyDagExtractor.boxed(),
+                is_dag_optimal: false,
+                is_tree_optimal: false,
+            },
+        ),
+        /*(
             "global-greedy-dag",
-            extract::global_greedy_dag::GlobalGreedyDagExtractor.boxed(),
+            ExtractorDetail {
+                extractor: extract::global_greedy_dag::GlobalGreedyDagExtractor.boxed(),
+                is_dag_optimal: false,
+                is_tree_optimal: false,
+            },
+        ),*/
+        #[cfg(feature = "ilp-cbc")]
+        (
+            "ilp-cbc",
+            ExtractorDetail {
+                extractor: extract::ilp_cbc::CbcExtractor.boxed(),
+                is_dag_optimal: true,
```

Comment:
Is this true? I thought we had no dag optimal algorithm.

Reply:
#30 and #16 both produce an optimal dag extraction given infinite time.

I can make this false until #30 is accepted? Or change the name to produces_dag_optimal_extractions?

Reply:
As far as I can tell, #30 also falls back on `faster-greedy-dag` after a time limit, thus not always producing optimal results.

I would love to have this check if it was actually true- maybe we should check if the ILP succeeded, and only do the check in that case?

Reply:
I'm happy to change the ILP extractors so that they return None if they timeout? That'd make it clear whether it's an optimal extraction (or not).

Reply:
That sounds like a good idea to me!

Reply:
In #30 there are two ILP extractors now, one which timeouts and one which doesn't. I've updated the code to use the ILP extractor with a timeout when doing the benchmarks, but to use it without the timeout when doing the fuzz testing.